MantidPlot/ipython_widget/mantid_ipython_widget.py

```diff
@@ -47,18 +47,20 @@ def our_run_code(self, code_obj, result=None):
           A compiled code object, to be executed
         result : ExecutionResult, optional
           An object to store exceptions that occur during execution.
+        async_ : Bool (Experimental))
+           Attempt to run top-level asynchronous code in a default loop.
         Returns
         -------
         False : Always, as it doesn't seem to matter.
     """
-
-    t = threading.Thread()
-    #ipython 3.0 introduces a third argument named result
-    nargs = len(inspect.getargspec(self.ipython_run_code).args)
-    if (nargs == 3):
-        t = threading.Thread(target=self.ipython_run_code, args=[code_obj,result])
+    # Different target arguments depending on IPython's version
+    if 'result' in getfullargspec(self.ipython_run_code).args:
+        if 'async_' in getfullargspec(self.ipython_run_code).kwonlyargs:
```

Comment:
It looks like this is not compatible with Python 2. I get the following error on trying to execute a line of code:
```python
AttributeError                            Traceback (most recent call last)
/home/mgigg/Code/mantidproject/mantid-upstream/qt/python/mantidqt/widgets/jupyterconsole.py in async_run_code(self, code_obj, result, async_)
     88         # Different target arguments depending on IPython's version
     89         if 'result' in getfullargspec(orig_run_code).args:
---> 90             if 'async_' in getfullargspec(orig_run_code).kwonlyargs:
     91                 return orig_run_code(code_obj, result, async_=async_)  # return coroutine to be awaited
     92             else:

AttributeError: 'ArgSpec' object has no attribute 'kwonlyargs'
```

Reply:
I forgot python 2.x doesn't have keyword-only arguments

Reply:
I am on a conference/vacation until next Monday. I doubt I will have enough time to work on this until then